Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -0,0 +1,564 @@
+        config_data[9] =
+            getNumberOfIntegerVariablesOfElements(partition.ghost_elements);
+
+        fwrite(config_data, 1, num_config_data * sizeof(IntegerType), of_bin_cfg);
+
+        config_data[10] += config_data[0] * sizeof(NodeStruct);
+
+        // Update offsets
+        num_elem_integers[loop_id] =
+            partition.regular_elements.size() + config_data[8];
+        // Offset the ending enrtry of the element integer variales of
+        // the non-ghost elements of this partition in the vector of elem_info.
+        config_data[11] += num_elem_integers[loop_id] * sizeof(IntegerType);
+        // Offset the ending enrtry of the element integer variales of
+        // the ghost elements of this partition in the vector of elem_info.
+        num_g_elem_integers[loop_id] =
+            partition.ghost_elements.size() + config_data[9];
+        config_data[12] += num_g_elem_integers[loop_id] * sizeof(IntegerType);
+
+        loop_id++;
+    }
+    return  std::make_tuple(num_elem_integers, num_g_elem_integers);
+}
+
+void NodeWiseMeshPartitioner::writeElementsBinary
+                     (const std::string& file_name_base,
+                      const std::vector<IntegerType>& num_elem_integers,
+                      const std::vector<IntegerType>& num_g_elem_integers)
+{
+    const std::string npartitions_str = std::to_string(_npartitions);
+    std::string fname = file_name_base + "_partitioned_msh_ele"
+                              + npartitions_str + ".bin";
+    FILE* of_bin_ele = fopen(fname.c_str(), "wb");
+    fname =
+        file_name_base + "_partitioned_msh_ele_g" + npartitions_str + ".bin";
+    FILE* of_bin_ele_g = fopen(fname.c_str(), "wb");
+    for (std::size_t i = 0; i < _partitions.size(); i++)
+    {
+        const auto& partition = _partitions[i];
+
+        // Set the local node indices of the current partition.
+        IntegerType node_local_id_offset = 0;
+        std::vector<IntegerType> nodes_local_ids(_mesh->getNumberOfNodes(), -1);
+        for (const auto* node : partition.nodes)
+        {
+            nodes_local_ids[node->getID()] = node_local_id_offset;
+            node_local_id_offset++;
+        }
+
+        // An vector contians all element integer variales of
+        // the non-ghost elements of this partition
+        std::vector<IntegerType> ele_info(num_elem_integers[i]);
+
+        // Non-ghost elements.
+        IntegerType counter = partition.regular_elements.size();
+
+        for (std::size_t j = 0; j < partition.regular_elements.size(); j++)
+        {
+            const auto* elem = partition.regular_elements[j];
+            ele_info[j] = counter;
+            getElementIntegerVariables(*elem, nodes_local_ids, ele_info,
+                                       counter);
+        }
+        // Write vector data of non-ghost elements
+        fwrite(&ele_info[0], 1, (num_elem_integers[i]) * sizeof(IntegerType),
```

Comment:
But a `std::vector` compliant with the c++ standard should have the correct `data()` method.
Furthermore: If that's a problem with some compiler, it will be a problem at a hundred places in our code. Still: change optional.